src/reference/asciidoc/xmpp.adoc

```diff
@@ -80,6 +80,42 @@ When started it will register a `PacketListener` that will listen for incoming X
 It forwards any received messages to the underlying adapter which will convert them to Spring Integration Messages and send them to the specified `channel`.
 It will unregister the `PacketListener` when it is stopped.
 
+Starting with _version 4.3_ the `ChatMessageListeningEndpoint` (and its `<int-xmpp:inbound-channel-adapter>`)
+supports a `org.jivesoftware.smack.filter.StanzaFilter` injection to be registered on the provided `XMPPConnection`
+together with an internal `StanzaListener` implementation.
+See their https://www.igniterealtime.org/builds/smack/docs/latest/javadoc/org/jivesoftware/smack/XMPPConnection.html#addAsyncStanzaListener%28org.jivesoftware.smack.StanzaListener,%20org.jivesoftware.smack.filter.StanzaFilter%29[JavaDocs] for more information.
+
+Also with the _version 4.3_ the `payload-expression` has been introduced for the `ChatMessageListeningEndpoint`.
+The incoming `org.jivesoftware.smack.packet.Message` represents a root object of evaluation context.
+This option is useful in case of <<xmpp-extensions>>.
+For example, for the GCM protocol we can extract the body using expression:
+
+[source,xml]
+----
+payload-expression="getExtension('google:mobile:data').json"
```

Comment:
I see a language thing here. My concern was more on the understanding side. As in the Spring REST the **.json** extension automatically triggers the Jackson mapper to produce the right result.

Just reading this code sample and assuming I have a Spml Expression my first thought was that a JSON conversion is done by this expression. Basically:
_Take Extension -> Do json mapping_

After remembering that the default property for the payload was "json" it made more sense, just getter access. Was just wondering if it would make sense to point it out in the docs or I was just to confused :-)